### PR TITLE
Increase coverage for generated chains and proptests

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

1. We always have 5 transactions in each generated block
2. I got the coinbase maturity limit confused with the legacy chain block limit while writing PR #2525

## Solution

- Allow 1-5 transactions in each generated block, so our tests cover near-empty blocks
- Make the legacy chain block limit into a constant, and use it in the tests

## Review

@oxarbitrage can review this PR, it's not urgent.

### Reviewer Checklist

  - [ ] Code works as documented
  - [ ] Existing tests still pass

---
This change is [Reviewable](https://reviewable.io/reviews/zcashfoundation/zebra/2540)

### CODE REVIEW

File: zebra-state/src/service/tests.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ use zebra_chain::{
 };
 use zebra_test::{prelude::*, transcript::Transcript};
 
-use crate::{init_test, tests::setup::partial_nu5_chain_strategy, BoxError, Request, Response};
+use crate::{constants, init_test, tests::setup::partial_nu5_chain_strategy, BoxError, Request, Response};
```
Comment (teor2345) [inline zebra-state/src/service/tests.rs]:
```suggestion
use crate::{
    constants, init_test, tests::setup::partial_nu5_chain_strategy, BoxError, Request, Response,
};
```

**Comment by dconnolly** [comment]
Where did the coverage job go? It's not uploading to codecov.io

**Comment by dconnolly** [comment]
> Where did the coverage job go? It's not uploading to codecov.io

We haven't reported coverage in over 20 days

image

**Comment by dconnolly** [comment]
> > Where did the coverage job go? It's not uploading to codecov.io
> 
> We haven't reported coverage in over 20 days
> 
> image

I figured it out: when we renamed the workflow to 'Coverage', there was previously an earlier iteration of that workflow with the same name that had been manually disabled in in the GitHub UI, to stop it appearing spuriously in several UI surfaces. Apparently that 'disabled' designation matches exactly on the name, not any other workflow id. 🤦‍♂️

I have re-enabled the workflow now.

**Comment by teor2345** [comment]
> I figured it out: when we renamed the workflow to 'Coverage', there was previously an earlier iteration of that workflow with the same name that had been manually disabled in in the GitHub UI, to stop it appearing spuriously in several UI surfaces. Apparently that 'disabled' designation matches exactly on the name, not any other workflow id. 🤦‍♂️
> 
> I have re-enabled the workflow now.

I disabled coverage because it was failing, and didn't need to be fixed urgently. (See #2454.)

I have disabled it again, because it's still failing.

**Review by oxarbitrage — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 1 of 1 files at r2, 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/zcashfoundation/zebra/2540)* status: all files reviewed, 1 unresolved discussion (waiting on @teor2345)

**Pull request merged**